mmhuman3d/data/data_structures/human_data.py

```diff
@@ -1056,6 +1074,131 @@ def __set_default_values__(self):
+        """
+        ret_human_data = cls.new(key_strict=False)
+        set_0 = set(human_data_0.keys())
+        set_1 = set(human_data_1.keys())
+        common_keys = set_0.intersection(set_1)
+        dim_dict_0 = human_data_0.__get_slice_dim__()
+        dim_dict_1 = human_data_1.__get_slice_dim__()
+        for key in common_keys:
+            value_0 = human_data_0[key]
+            value_1 = human_data_1[key]
+            # align type
+            value_0 = list(value_0) if isinstance(value_0, tuple)\
+                else value_0
+            value_1 = list(value_1) if isinstance(value_1, tuple)\
+                else value_1
+            assert type(value_0) == type(value_1)
+            # align convention
+            if key.startswith('keypoints') and\
+                    key.endswith('_convention'):
+                assert value_0 == value_1
+                ret_human_data[key] = value_0
+                continue
+            # mask_0 and mask_1
+            elif key.startswith('keypoints') and\
+                    key.endswith('_mask'):
+                new_mask = value_0 * value_1
+                ret_human_data[key] = new_mask
+                continue
+            # go through the sub dict
+            if isinstance(value_0, dict):
+                sub_dict = {}
+                for sub_key, sub_value_0 in value_0.items():
+                    # only found in value_0
+                    if sub_key not in value_1:
+                        sub_dict[sub_key] = sub_value_0
+                    # found in both values
+                    else:
+                        sub_value_1 = value_1[sub_key]
+                        concat_sub_dict = cls.__concat_value__(
+                            key=sub_key,
+                            value_0=sub_value_0,
+                            dim_0=dim_dict_0[key][sub_key],
+                            value_1=sub_value_1,
+                            dim_1=dim_dict_1[key][sub_key])
+                        sub_dict.update(concat_sub_dict)
+                for sub_key, sub_value_1 in value_1.items():
+                    if sub_key not in value_0:
+                        sub_dict[sub_key] = sub_value_1
+                ret_human_data[key] = sub_dict
+            # try concat
+            else:
+                concat_dict = cls.__concat_value__(
+                    key=key,
+                    value_0=value_0,
+                    dim_0=dim_dict_0[key],
+                    value_1=value_1,
+                    dim_1=dim_dict_1[key])
+                ret_human_data.update(concat_dict)
+        # check exclusive keys
+        for key, value in human_data_0.items():
+            if key not in common_keys:
+                # value not for concat and slice
+                if dim_dict_0[key] is None:
+                    ret_human_data[key] = value
+                # value aligned with data_len of HumanData_0
+                else:
+                    ret_human_data[f'{key}_0'] = value
```

Comment:
There are two cases:
1. human_data_0 has `betas` of (100, 10) in `smpl` sub-dict, while human_data_1 has `betas` of (1, 10) in `smpl`, which does not align with `data_len`. We will keep both of them instead of concat.
2. human_data_0 has `key_frame_mask` of (100, ), `ndarray`, while human_data_1 has `key_frame_mask` of (100, ), `list`. As `key_frame_mask` is not supported, we do not have expected type for the key, but the data_len is accpeted. The only thing we could do is to take both values and mark their sources.

Considering the warning strategy and auto slice function for unsupported keys, I treat HumanData as a tolerant class. It works well for slicing itself into a HumanDataCache and concatenating them back to HumanData, and do not crash for small mistakes.

Reply:
> 1. human_data_0 has `betas` of (100, 10) in `smpl` sub-dict, while human_data_1 has `betas` of (1, 10) in `smpl`, which does not align with `data_len`. We will keep both of them instead of concat.

Shouldn't the concatenated `betas` have shape (101, 10)?

> 2. human_data_0 has `key_frame_mask` of (100, ), `ndarray`, while human_data_1 has `key_frame_mask` of (100, ), `list`. As `key_frame_mask` is not supported, we do not have expected type for the key, but the data_len is accpeted. The only thing we could do is to take both values and mark their sources.

If both human_data contains an unsupported key but the same value type (assuming np.array) and can be concatenated, should we still try to concatenate them? Or a proper way is to add these keys to supported keys?

Reply:
For human_data_1 in the first question, user use `betas` (1, 10) standing for 100 frames, the expected concatenated betas is (200, 10). Obviously, the repeat operation is beyond HumanData's ability and knowledge. When `global_orient` is concatenated to (200, 3), a (101, 10) betas is not accpeted, and another person doesn't know where to split it.

Reply:
For the second question, if any key is not defined, but we find it points to two correct values that has the same type, and dimension 0 means data_len, we will concatenate them because that we are confident to the result. If types or meaning sof dim 0 dis-match, there's no need to try. The user shall forget about what type is the origin value and where to split in ten minutes.